### DISCUSSION TITLE
Locale robots.txt and sitemap.xml

### DISCUSSION THREAD

**Discussion by levipadre** [amannn/next-intl#264] [Q&A] [answered]
I'm doing a project using Next.js version 13 with beta app directory. I'm hosting on Vercel.

I installed next-intl package to make my project international, and I can use different domain names.
My file structure under the app folder is /[locale]/page.js and /[locale]/layout.js etc.
I want to add a robots file under each locales but if I place the following robots.js file under the [locale] folder, it it doesn't work.
If I place it just under the app folder, it works, but in this case, I can't generate the domain name for the sitemap value, so no locale.
This is my robots.js file:
```
export default function robots() {
    return {
        rules: {
            userAgent: '*',
            allow: '/'
        },
        sitemap: 'https://domain.com/sitemap.xml',
    };
}
```

Same situation with sitemap.js file.

What shall I do to provide different robots.txt and sitemap.xml for the different domain names?

**Comment by amannn** [comment, marked as answer]
That's an interesting question! I think your best shot is to read the host from the incoming request:

```tsx
const host = headers().get('x-forwarded-host') || headers().get('host');
```

Hope this helps!

**Reply by levipadre** [reply]
Thank you so much, @amannn. It's perfect.
I was also wondering if it's possible to add additional information to middleware, like a custom id so that I can get the information somehow. From headers() maybe? I'm using 2.14.0-beta.3 version.
Like:
```
domains: [
        {
            domain: 'domain.com',
            defaultLocale: 'en',
            locales: ['en'],
            id: 123
        },
        {
            domain: 'domain.fr',
            defaultLocale: 'fr',
            locales: ['fr'],
            id: 987
        },
 ]
```
